crates/bevy_sprite/src/render/mod.rs

```diff
@@ -233,23 +240,49 @@ pub fn extract_sprites(
 ) {
     let mut extracted_sprites = render_world.resource_mut::<ExtractedSprites>();
     extracted_sprites.sprites.clear();
-    for (visibility, sprite, transform, handle) in sprite_query.iter() {
+    for (visibility, sprite, transform, handle, slicer) in sprite_query.iter() {
         if !visibility.is_visible {
             continue;
         }
-        // PERF: we don't check in this function that the `Image` asset is ready, since it should be in most cases and hashing the handle is expensive
-        extracted_sprites.sprites.alloc().init(ExtractedSprite {
-            color: sprite.color,
-            transform: *transform,
-            // Use the full texture
-            rect: None,
-            // Pass the custom size
-            custom_size: sprite.custom_size,
-            flip_x: sprite.flip_x,
-            flip_y: sprite.flip_y,
-            image_handle_id: handle.id,
-            anchor: sprite.anchor.as_vec(),
-        });
+        if let Some(slicer) = slicer {
```

Comment:
I don't love introducing more complexity + a branch into a hot-path system like this. I'd be tempted to benchmark this as one system vs two.

Reply:
https://github.com/bevyengine/bevy/pull/5213#discussion_r914906314

would remove the extra component